frontend/src/components/AuthUtils/index.js

```diff
@@ -0,0 +1,65 @@
+/**
+ * Utility functions to get various parameters corresponding to the currently
+ * authenticated user. These functions should ONLY be called from private pages,
+ * i.e. the user must be authenticated to ensure the parameters can be
+ * retrieved.
+ */
+
+import { AuthUtilsConsumer, getCurrentUser } from './AuthUtilsConsumer.js';
+import { SIGN_IN } from '../../constants/routes.js';
+
+/**
+ * Called when one of the authentication utility functions are used when the
+ * user has not logged in yet. Since the utility functions require that they be
+ * called from private pages, this function being called implies that the user
+ * managed to access a private page without being authenticated first. So they
+ * are immiedately redirected to the SIGN_IN page.
+ */
+function redirectToSignIn() {
+  window.location.href = SIGN_IN;
+}
+
+export function getUserDisplayName() {
+  if (!getCurrentUser()) {
+    redirectToSignIn();
+    return null;
+  }
+  return getCurrentUser().displayName;
+}
+
+export function getUserEmail() {
+  if (!getCurrentUser()) {
+    redirectToSignIn();
+    return null;
+  }
+  return getCurrentUser().email;
+}
+
+export function getUserPhotoUrl() {
+  if (!getCurrentUser()) {
+    redirectToSignIn();
+    return null;
+  }
+  return getCurrentUser().photoURL;
+}
+
+export function getUserUid() {
+  if (!getCurrentUser()) {
+    redirectToSignIn();
+    return null;
+  }
+  return getCurrentUser().uid;
+}
+
+// Can also access the auth functions in the named authUtils variable.
+const authUtils = {
+  getUserDisplayName,
+  getUserEmail,
+  getUserPhotoUrl,
+  getUserUid
+};
```

Comment:
These functions look good for grabbing info about the current user. But I think there may be a gap between this and what I need for trips. 

Based on our discussion of integration between Auth and Trip management, I assumed that I can get a uid from any user email and vice versa on the view trips page. This allows me to get collaborator uids so I can store those in firestore given the emails the trip creator provided. Same thing when fetching trips but in the reverse order (getting the emails to display on the screen from the stored uids.

Do you think you can implement this soon in a future PR? If this doesn't sound feasible, then we may need to chat about the core design again.

Reply:
Yeah I'm still trying to find the best way to do this for my end, since I'll likely need to use the Firebase Admin SDK and _may_ need to use to do that in a Java Servlet. I do plan on implementing it soon in a future PR, hopefully by the end of this week. Not sure if you're fine with using the getUserEmail function for now until that's implemented?

Reply:
Okay great. Yeah I use the getUserEmail function to get the current user email in addition to the two I mentioned above so that should work great. Let me know if you need help with anything and then I will integrate those functions once you got them figured out.

Reply:
Cool, thanks!